diff --git a/Sources/CLibMongoC/mongoc/mongoc-topology-private.h b/Sources/CLibMongoC/mongoc/mongoc-topology-private.h
index 350b3b5..0e8ea28 100644
--- a/Sources/CLibMongoC/mongoc/mongoc-topology-private.h
+++ b/Sources/CLibMongoC/mongoc/mongoc-topology-private.h
@@ -28,7 +28,7 @@
 #include "mongoc-client-session-private.h"
 #include "mongoc-crypt-private.h"
 
-#define MONGOC_TOPOLOGY_MIN_HEARTBEAT_FREQUENCY_MS 500
+#define MONGOC_TOPOLOGY_MIN_HEARTBEAT_FREQUENCY_MS 50
 #define MONGOC_TOPOLOGY_SOCKET_CHECK_INTERVAL_MS 5000
 #define MONGOC_TOPOLOGY_COOLDOWN_MS 5000
 #define MONGOC_TOPOLOGY_LOCAL_THRESHOLD_MS 15
