diff --git a/Sources/CLibMongoC/include/CLibMongoC_mongoc-client-pool.h b/Sources/CLibMongoC/include/CLibMongoC_mongoc-client-pool.h
index 16d7eb1..bcf1cde 100644
--- a/Sources/CLibMongoC/include/CLibMongoC_mongoc-client-pool.h
+++ b/Sources/CLibMongoC/include/CLibMongoC_mongoc-client-pool.h
@@ -76,6 +76,10 @@ mongoc_client_pool_set_server_api (mongoc_client_pool_t *pool,
                                    const mongoc_server_api_t *api,
                                    bson_error_t *error);
 
+MONGOC_EXPORT (void)
+mongoc_client_pool_set_min_heartbeat_frequency_msec (mongoc_client_pool_t *pool,
+                                                     uint64_t freq_msec);
+
 BSON_END_DECLS
 
 
diff --git a/Sources/CLibMongoC/mongoc/mongoc-client-pool.c b/Sources/CLibMongoC/mongoc/mongoc-client-pool.c
index ea78c7a..2139330 100644
--- a/Sources/CLibMongoC/mongoc/mongoc-client-pool.c
+++ b/Sources/CLibMongoC/mongoc/mongoc-client-pool.c
@@ -543,3 +543,10 @@ mongoc_client_pool_set_server_api (mongoc_client_pool_t *pool,
    _mongoc_topology_scanner_set_server_api (pool->topology->scanner, api);
    return true;
 }
+
+void
+mongoc_client_pool_set_min_heartbeat_frequency_msec (mongoc_client_pool_t *pool,
+                                                     uint64_t freq_msec)
+{
+    pool-> topology-> min_heartbeat_frequency_msec = freq_msec;
+}
diff --git a/Sources/CLibMongoC/mongoc/mongoc-topology-private.h b/Sources/CLibMongoC/mongoc/mongoc-topology-private.h
index 445468b..be181b9 100644
--- a/Sources/CLibMongoC/mongoc/mongoc-topology-private.h
+++ b/Sources/CLibMongoC/mongoc/mongoc-topology-private.h
@@ -28,7 +28,7 @@
 #include "mongoc-client-session-private.h"
 #include "mongoc-crypt-private.h"
 
-#define MONGOC_TOPOLOGY_MIN_HEARTBEAT_FREQUENCY_MS 500
+#define MONGOC_TOPOLOGY_MIN_HEARTBEAT_FREQUENCY_MS 50
 #define MONGOC_TOPOLOGY_SOCKET_CHECK_INTERVAL_MS 5000
 #define MONGOC_TOPOLOGY_COOLDOWN_MS 5000
 #define MONGOC_TOPOLOGY_LOCAL_THRESHOLD_MS 15
